demos/common/cpp/utils/include/utils/images_capture.h

```diff
@@ -8,12 +8,26 @@
 #include <opencv2/core/mat.hpp>
 #include <opencv2/videoio.hpp>
 
+
+class InvalidInput : public std::runtime_error {
+public:
+    explicit InvalidInput(const std::string& message) noexcept
+        : std::runtime_error(message) {}
+};
+
+class OpenError : public std::runtime_error {
+public:
+    explicit OpenError(const std::string& message) noexcept
+        : std::runtime_error(message) {}
+};
```

Comment:
These two classes shouldn't be visible. There is no way for them to be thrown from anything visible from the header. Move them into cpp file.